packages/vaadin-time-picker/src/vaadin-time-picker.js

```diff
@@ -382,42 +288,104 @@ class TimePickerElement extends ElementMixin(ControlStateMixin(ThemableMixin(Pol
                 }
                 return { hours: parts[1], minutes: parts[2], seconds: parts[3], milliseconds: parts[4] };
               }
-            },
-            selector: 'Time selector',
-            clear: 'Clear'
+            }
           };
         }
-      }
+      },
+
+      /** @private */
+      _comboBoxValue: {
+        type: String,
+        observer: '__comboBoxValueChanged'
+      },
+
+      /** @private */
+      _inputContainer: Object
     };
   }
 
   static get observers() {
     return ['__updateDropdownItems(i18n.*, min, max, step)'];
   }
 
+  /**
+   * Element used by `FieldAriaMixin` to set ARIA attributes.
+   * @protected
+   */
+  get _ariaTarget() {
+    return this.inputElement;
+  }
+
+  /**
+   * Used by `ClearButtonMixin` as a reference to the clear button element.
+   * @protected
+   * @return {!HTMLElement}
+   */
+  get clearElement() {
+    return this.$.clearButton;
```

Comment:
Pressing esc should also clear the value. Not a regression in this PR though and could be fixed separately.

Reply:
Reported separately in https://github.com/vaadin/web-components/issues/2614 This doesn't seem to be a recent regression; at least the combo-box in https://vaadin.com/docs/v20/ds/components/combo-box has the same issue